src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -288,11 +294,16 @@ function createObjectOfTypeChecker(typeChecker) {
 
 function createUnionTypeChecker(arrayOfTypeCheckers) {
   if (!Array.isArray(arrayOfTypeCheckers)) {
-    return createChainableTypeChecker(function() {
-      return new Error(
-        `Invalid argument supplied to oneOfType, expected an instance of array.`
-      );
-    });
+    if (__DEV__) {
+      warning(false, 'Invalid argument supplied to oneOfType, expected an instance of array.');
+      return emptyFunction.thatReturnsNull;
+    } else {
+      return createChainableTypeChecker(function() {
```

Comment:
Same here. So basically both of these blocks should become

```js
warning(...)
return emptyFunction.thatReturnsNull;
```

`__DEV__` check will be added automatically when you call `warning`.